### PR TITLE
Documentation for scenario file format

### LINKED ISSUE #354: Documentation for how to author scenarios
This is potentially related to #346 and #347, but is fundamentally separate.  We need some documentation somewhere (e.g. on the [wiki](https://github.com/swarm-game/swarm/wiki)?) explaining the format of the `.yaml` files used to describe scenarios and what all the different options mean.

### PR DESCRIPTION
**Pull request by byorgey**
Closes #354 .

### CODE REVIEW

File: data/scenarios/README.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+| `creative`     | `False`  | `boolean`  | Whether the scenario should start out in creative mode.                                                                                                                                                                                                                                                                                                       |
+| `seed`         | `null`   | `int`      | An optional seed that will be used to seed the random number generator.  If a procedurally generated world is used, the seed hence determines the world.  Hence, if the seed is specified, the procedurally generated world will be exactly the same every time, for every player.  If omitted, a random seed will be used every time the scenario is loaded. |
+| `entities`     | `[]`     | `map list` | An optional list of custom entities, to be used in addition to the built-in entities.  See [Entities](#entities).                                                                                                                                                                                                                                             |
```
Comment (xsebek) [inline data/scenarios/README.md:62]: The `map list` is not rendered very nicely in README and is IMHO too confusing.

I would suggest using:
```markdown
[`entity`](#entites) list
```

File: data/scenarios/README.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+| `description`  |          | `string list` | A description of the entity, as a list of paragraphs.                                                                                                                                                                                                                            |
+| `orientation`  | `null`   | `int × int`   | A 2-tuple of integers specifying an orientation vector for the entity. Currently unused.                                                                                                                                                                                         |
+| `growth`       | `null`   | `int × int`   | For growable entities, a 2-tuple of integers specifying the minimum and maximum amount of time taken for one growth stage.  The actual time for one growth stage will be chosen uniformly at random from this range; it takes two growth stages for an entity to be fully grown. |
```
Comment (xsebek) [inline data/scenarios/README.md:83]: AFAIK, we use a heterogenous list for pair and check that it in this case contains two entries that are both integers.

I think that needs an explanation somewhere. :confused:

File: data/scenarios/README.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+creative mode, tutorial, challenges, *etc.*) are described via
+scenarios. Scenarios are stored in human-friendly `.yaml` files, so it
+is very easy---and encouraged!---for you to create your own scenarios
+representing puzzles, challenges, works of art, or whatever you want.
+This page formally documents the `.yaml` format and also provides some
+tips for authoring your own scenarios.
+
+If you're impatient to get started, note that a much quicker route is
+to look at the scenario files stored in
+https://github.com/swarm-game/swarm/tree/main/data/scenarios , copy
```
Comment (xsebek) [inline data/scenarios/README.md:15]: That extra space ... :sweat_smile: 
```suggestion
 [`data/scenarios`](https://github.com/swarm-game/swarm/tree/main/data/scenarios) of the swarm repository, copy
```

**Review by xsebek — APPROVED** [review]
Looks good to me! :+1: Hopefully it will be a good enough explanation for future writers. :slightly_smiling_face:

File: data/scenarios/README.md (hunk 2 above)
Comment (byorgey) [reply data/scenarios/README.md:97]: Good point!  I added a section explaining some conventions.

**Comment by byorgey** [comment]
> Hopefully it will be a good enough explanation for future writers.

Hopefully so, and if not, they can ask questions or make a PR. :smile:

**Comment by byorgey** [comment]
Is this one of those scenarios where the tests will never run since this PR only touched an .md file?

**Comment by byorgey** [comment]
I just merged manually.

**Comment by xsebek** [comment]
Yeah, we need to look into the rules, so that mergify could handle it. 🙂

**Pull request merged**